Adding inn-interface to conda-forge

jefalon opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with a PR, please let the
right people know. There are language-specific teams for reviewing recipes.

| Language        | Name of review team           |
| --------------- | ----------------------------- |
| python          | `@conda-forge/help-python`    |
| python/c hybrid | `@conda-forge/help-python-c`  |
| r               | `@conda-forge/help-r`         |
| java            | `@conda-forge/help-java`      |
| nodejs          | `@conda-forge/help-nodejs`    |
| c/c++           | `@conda-forge/help-c-cpp`     |
| perl            | `@conda-forge/help-perl`      |
| Julia           | `@conda-forge/help-julia`     |
| ruby            | `@conda-forge/help-ruby`      |
| other           | `@conda-forge/staged-recipes` |

Once the PR is ready for review, please mention one of the teams above in a
new comment. i.e. `@conda-forge/help-some-language, ready for review!`
Then, a bot will label the PR as 'review-requested'.

Due to GitHub limitations, first time contributors to conda-forge are unable
to ping conda-forge teams directly, but you can [ask a bot to ping the team][1]
using a special command in a comment on the PR to get the attention of the
`staged-recipes` team. You can also consider asking on our [Gitter channel][2]
if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://gitter.im/conda-forge/conda-forge.github.io
-->

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/inn-interface```) and found it was in an excellent condition.


I do have some suggestions for making it better though...


For **recipes/inn-interface**:

 * Recipes should usually depend on `matplotlib-base` as opposed to `matplotlib` so that runtime environments do not require large packages like `qt`.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/inn-interface```) and found some lint.

Here's what I've got...


For **recipes/inn-interface**:

 * `noarch` packages can't have selectors. If the selectors are necessary, please remove `noarch: python`.

For **recipes/inn-interface**:

 * Recipes should usually depend on `matplotlib-base` as opposed to `matplotlib` so that runtime environments do not require large packages like `qt`.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/inn-interface```) and found some lint.

Here's what I've got...


For **recipes/inn-interface**:

 * Non noarch packages should have python requirement without any version constraints.
 * Non noarch packages should have python requirement without any version constraints.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/inn-interface```) and found it was in an excellent condition.

jefalon commented:
https://github.com/orgs/conda-forge/teams/help-python Ready for review!

jefalon commented:
@conda-forge/help-python Ready for review!

jhkennedy commented on recipes/inn-interface/meta.yaml:
```diff
@@ -0,0 +1,53 @@
+    - INN_interface
+
+about:
+  home: https://https://github.com/NREL/INN-interface
```
```suggestion
  home: https://github.com/NREL/INN-interface
```

jefalon replied:
Oh whoops, thanks for catching that!

jhkennedy reviewed (changes requested):
This looks good to me, just needs one small change.

jhkennedy commented on recipes/inn-interface/meta.yaml:
```diff
@@ -0,0 +1,53 @@
+
+extra:
+  recipe-maintainers:
+    - aglaws
```
@jefalon since you submitted this recipe, did you want to be listed as a maintainer too?

jefalon replied:
I'm just helping Andrew get this package onto conda-forge since I have experience doing it in the past, so I don't need to be a maintainer.

jhkennedy reviewed (approved):
Looks good to me! Just one question that's not blocking

carterbox commented on recipes/inn-interface/meta.yaml:
```diff
@@ -0,0 +1,53 @@
+    - tensorflow >=2.7.0
+    - wisdem
+
+test:
+  imports:
+    - INN_interface
```
```suggestion
test:
  requires:
    - pip
  commands:
    - pip check
  imports:
    - INN_interface
```

carterbox reviewed (changes requested):
If the only reason this recipe isn't noarch is that tensorflow isn't available, then please revert back to noarch. Unavailable dependencies are for noarch because the package will just be uninstallable on that platform.

jefalon commented:
Hey @carterbox, I reverted to back noarch since the tensorflow dependency was the only reason I removed it in the first place. Let me know if this PR is now good to go!

jefalon commented:
@conda-forge/help-python I think this PR is ready for review again. I know the windows version is failing, but I was told to use noarch rather than skipping windows since it is just a dependency issue. If that is not the case, please let me know and I'll do what needs to get done! Thanks!

carterbox commented on recipes/inn-interface/meta.yaml:
```diff
@@ -0,0 +1,56 @@
+
+requirements:
+  host:
+    - python >=3.6
```
https://github.com/NREL/INN-interface/blob/ab4be7cdae5c1d1329c1e78636c0556eaa512f74/setup.py#L13

jefalon replied:
Hey @carterbox, I just checked and the files in examples and models folder are not needed for the module. I also updated the python version to match the setup.py. Thanks for catching that!

carterbox reviewed (changes requested):
According to [the logs](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=543049&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1080), none of the source files from the examples and models folders are being included because they are data and not python modules. Is that what you want?

jefalon commented:
@conda-forge/help-python Unless there is anything else that I need to change, this should be ready to go. Third times a charm! Thanks again for all your help getting this ready!
